backend/pkg/kubeconfig/kubeconfig.go

```diff
@@ -258,357 +279,492 @@ func (c *Context) AuthType() string {
+}
 
-	if err := clientcmd.Validate(*config); err != nil {
-		return rawConfig, config, err
+// GetContextsFromKubeconfig gets the contexts from the kubeconfig.
+func GetContextsFromKubeconfig(kubeconfig map[string]interface{}) ([]interface{}, error) {
+	rawContexts, ok := kubeconfig["contexts"].([]interface{})
+	if !ok {
+		return nil, DataError{Field: "contexts", Reason: "invalid or missing contexts in kubeconfig"}
 	}
 
-	return rawConfig, config, nil
+	return rawContexts, nil
 }
 
-// createContext creates a context from the given raw context, clusters, and users.
-// It returns the created context and any errors that occurred.
-// It returns an error if the context cannot be created.
-// It returns a valid context if the context is created.
-// It returns any errors that occurred during context creation.
-func createContext(rawContext interface{}, rawClusters, rawUsers []interface{}, source int) (Context, error) {
-	contextMap, ok := rawContext.(map[interface{}]interface{})
-	if !ok {
-		return Context{}, ContextError{ContextName: "unknown", Reason: "invalid context format"}
-	}
+// ProcessContext processes a context from the kubeconfig.
+// It returns errors for invalid contexts, but will return all contexts.
+// rawContext can be a single context or a list of contexts.
+// kubeconfig is the kubeconfig data.
+// source is the source of the kubeconfig, i.e where the kubeconfig came from.
+// It can be KubeConfig, DynamicCluster, or InCluster.
+// skipProxySetup is a flag to skip proxy setup.
+func ProcessContext(
+	rawContext interface{},
+	kubeconfig map[string]interface{},
+	source int,
+	skipProxySetup bool,
+) (Context, error) {
+	var errs []error
 
-	name, _ := contextMap["name"].(string)
-	contextData, _ := contextMap["context"].(map[interface{}]interface{})
-	clusterName, _ := contextData["cluster"].(string)
-	userName, _ := contextData["user"].(string)
+	var context Context
+	// Extract context information
+	contextMap, contextName, err := extractContextInfo(rawContext)
+	if err != nil {
+		errs = append(errs, err)
+	}
 
-	cluster, err := findCluster(rawClusters, clusterName)
+	// Extract cluster and user names
+	clusterName, userName, err := extractClusterAndUserNames(contextMap, contextName)
 	if err != nil {
-		return Context{}, ContextError{ContextName: name, Reason: err.Error()}
+		errs = append(errs, err)
 	}
 
-	authInfo, err := findAuthInfo(rawUsers, userName)
+	// Get the cluster and user details
+	cluster, user, err := getClusterAndUser(kubeconfig, clusterName, userName)
 	if err != nil {
-		return Context{}, ContextError{ContextName: name, Reason: err.Error()}
+		errs = append(errs, err)
 	}
 
-	friendlyName := makeDNSFriendly(name)
-	kubeContext := createKubeContext(contextData)
+	// Create and validate the config
+	singleConfig, err := createAndValidateConfig(contextName, contextMap, cluster, user, kubeconfig)
+	if err != nil {
+		errs = append(errs, err)
+		context = Context{
+			Name:  contextName,
```

Comment:
I think Name here is not unique. So where there's non-unique names finding which error goes with which context won't be possible.